proposals/operator-ingress-sketch.md

```diff
@@ -0,0 +1,189 @@
+# Operator Ingress API sketch
+
+- I want simple things to be easy, harder things should still to be possible.
+- I want to be able to get a usable proxy up and running using as little yaml as possible.   I value
+  this because I thik it give a compelling demo that people can follow.
+
+## Roles
+
+* Infrastructure admin - responsible for the proxy tier and how it exposes itself to the network.
+* Developer - responsible for declaraction of the virtualcluster
+
+## API
+
+### Proxy
+
+Owned by the Infrastructure admin.
+
+Declares an instance of the proxy.  It is valid for this to be an empty object.  The empty object
```

Comment:
> It is valid for this to be an empty object.

I'm not sure this is really something worth striving for. The more you rely on defaulting the more cryptic things look to someone who doesn't know the API. While a single replica is a fairly guessable default I'm not so sure that an uninformed reader would guess "If no ingresses are defined, the operator will behave as if a plain clusterip ingress is defined called default."

Reply:
Let me explain my motivation. I want to be able to stand up for a demo and say "let me show you how you add encryption at rest to your kafka" and then proceed to do it with as few lines of yaml as possible - something that people can hold in their heads.  I figured that a proxy using plain protocol, exposed by services for on-cluster traffic, it is the simplest useful thing. I've tried to make that the easy thing whilst supporting more complex ingress patterns.

Maybe what I am suggesting here is not the right realisation of the idea.   I'm happy to keep iterating to get to something less surprising.

Reply:
I stepped away from this direction.